Tasks/Common/azure-blobstorage-downloader/azureBlobStorageProvider.ts

```diff
@@ -0,0 +1,139 @@
+                    reject(error);
+                } else {
+                    var blobUrl = self._blobSvc.getUrl(self._container, blobPath);
+                    console.log("Created blob for item " + item.path + ". Blob uri: " + blobUrl);
+                    newArtifactItem.metadata["downloadUrl"] = blobUrl;
+                    resolve(newArtifactItem);
+                }
+            });
+
+            readStream.pipe(writeStream);
+            writeStream.on("error",
+                (error) => {
+                    reject(error);
+                });
+            readStream.on("error",
+                (error) => {
+                    reject(error);
+                });
+        });
+    }
+
+    public getRootItems(): Promise<models.ArtifactItem[]> {
+        return new Promise(async (resolve, reject) => {
+            var items: models.ArtifactItem[] = await this._getItems(this._container)
+            resolve(items);
+        });
+    }
+
+    public getArtifactItems(artifactItem: models.ArtifactItem): Promise<models.ArtifactItem[]> {
+        return new Promise(async (resolve, reject) => {
+            var items: models.ArtifactItem[] = await this._getItems(this._container, artifactItem.path);
+            resolve(items);
+        });
+    }
+
+    public getArtifactItem(artifactItem: models.ArtifactItem): Promise<Stream.Readable> {
+        return new Promise(async (resolve, reject) => {
+            var readStream: Stream.Readable = this._blobSvc.createReadStream(this._container, artifactItem.path, (error, result) => {
+                if (error) {
+                    console.log("Unable to fetch item: " + artifactItem.path + ". Error: " + error);
+                    reject();
+                }
+            });
+            resolve(readStream);
+        });
+    }
+
+    private _ensureContainerExistence(): Promise<void> {
+        return new Promise<void>(async (resolve, reject) => {
+            if (!this._isContainerExists) {
+                var self = this;
+                this._blobSvc.createContainerIfNotExists(this._container, function (error, result, response) {
+                    if (!!error) {
+                        console.log("Failed to create container " + self._container + ". Error: " + error.message);
+                        reject(error);
+                    } else {
+                        self._isContainerExists = true;
+                        console.log("Created container " + self._container);
+                        resolve();
+                    }
+                });
+            } else {
+                resolve();
+            }
+        });
+    }
+
+    private _getItems(container: string, parentRelativePath?: string): Promise<models.ArtifactItem[]> {
+        var promise = new Promise<models.ArtifactItem[]>(async (resolve, reject) => {
+            var items: models.ArtifactItem[] = [];
+            await this._ensureContainerExistence();
```

Comment:
why do you need to ensure constainer existence?

Reply:
you are right, its not needed. removing it.